src/scmrepo/git/backend/pygit2.py

```diff
@@ -312,6 +312,9 @@ def resolve_commit(self, rev: str) -> "GitCommit":
             [str(parent) for parent in commit.parent_ids],
         )
 
+    def remove(self, paths: Union[str, Iterable[str]], **kwargs):
+        raise NotImplementedError
```

Comment:
Would be good to implement this here (porcelain `git rm` equivalent is index remove + local fs remove if the file actually exists in the workspace)

https://www.pygit2.org/index_file.html#pygit2.Index.remove

Reply:
I originally implemented it that way but found some issues. 

When testing as follows:

```
def test_remove(tmp_dir: TmpDir, scm: Git, git: Git):
    tmp_dir.gen({"foo": "foo", "goo": "goo"})
    # scm because pygit2 doesn't implement add/commit
    scm.add(["foo", "goo"])
    scm.commit("add files")
    git.remove(["foo", "goo"])
    staged, *_ = scm.status()
    assert staged["delete"] == ["foo", "goo"]
    assert not (tmp_dir / "foo").exists()
    assert not (tmp_dir / "goo").exists()
```

The `index.remove` implementation failed to pass the test.
I didn't look in depth at why it was the case (should do it if we really want to have `rm` in `pygit2` backend) because I just wanted `rm` in at least 1 backend.

I am not sure if it's just some issue with mixing the `scm` fixtures or if there as deeper problem

Reply:
One thing I can think of would be that you need to make sure you also call `index.write()` before returning from the pygit2 remove implementation, otherwise other backends won't see the changes (you can see the pygit2 backend implementations for merge and reset for examples)

Reply:
Indeed was missing the `write`